### ISSUE TITLE
[Proposal] Upgrade Notifications

### ISSUE THREAD

**Issue by rtablada** [wardrobecms/core-archived#1]
For mass appeal (and even niche appeal), Wardrobe CMS could benefit from having a notification of when Wardrobe/Core or Laravel/Framework are behind in a minor revision.

I am working on a package we should be able to integrate easily to allow for version comparison and manipulation.

From there we can build an API to check against when a user accesses the admin panel.

**Comment by ericlbarnes** [comment]
Artisan::call('composer update') then it could be automated :smile:

**Comment by rtablada** [comment]
Yeah, but I don't want to be running composer update every time a user logins.

If you aren't running on dev-master, I should be able to query the packagist API for available versions. If the version in the your composer is behind in non-breaking changes then you it will notify the user. The click to verify and it goes in, updates the composer dependency to the most recent minor/patch.

This may need to wait till we get out of RC since a comparator may not work and autoupdating while there may be breaking changes won't be too pretty...

**Comment by ericlbarnes** [comment]
Well I was meaning you would still need a version comparison and an alert. But they could just hit the upgrade button from the alert which them runs the composer update. So nothing is ever manual. I was have way joking because I don't believe that will be possible on shared hosts.

**Comment by rtablada** [comment]
Not sure about shared hosts. I'm looking into making a version control wrapper. I'll let you know how things are going. I'm looking at the composer source to see if I don't have to reinvent the wheel.
